internal/cmd/kubernetes_test.go

```diff
@@ -48,123 +43,123 @@ func TestWriteKubeconfig(t *testing.T) {
-				CertificateAuthorityData: []byte(destinationCA),
-			},
-		},
-		Contexts: map[string]*clientcmdapi.Context{
-			"infra:connected": {
-				AuthInfo: "user",
-				Cluster:  "infra:connected",
-			},
-		},
-		AuthInfos: map[string]*clientcmdapi.AuthInfo{
-			"user": {},
-		},
-	}
+		err := writeKubeconfig(&user, destinations, grants)
+		assert.NilError(t, err)
 
-	configFileStats, err := os.Stat(kubeConfigPath)
-	assert.NilError(t, err)
+		configFileStat, err := os.Stat(kubeConfigPath)
+		assert.NilError(t, err)
+		assert.Equal(t, int(configFileStat.Mode().Perm()), 0o600)
 
-	permissions := configFileStats.Mode().Perm()
-	assert.Equal(t, int(permissions), 0o600) // kube config should not be world or group readable, only user read/write
+		kubeConfig, err := clientConfig().RawConfig()
+		assert.NilError(t, err)
 
-	actual, err := clientConfig().RawConfig()
-	assert.NilError(t, err)
+		return kubeConfig
+	}
+
+	expectedClusters := map[string]*clientcmdapi.Cluster{
+		"infra:connected": {
+			Server:                   "https://connected.example.com",
+			CertificateAuthorityData: []byte(destinationCA),
+		},
+	}
 
-	assert.DeepEqual(t, expected, actual,
+	expectedAuthInfos := map[string]*clientcmdapi.AuthInfo{
+		"user": {},
+	}
+
+	kubeConfigCmpOpts := []cmp.Option{
 		cmpopts.EquateEmpty(),
-		cmpopts.IgnoreFields(clientcmdapi.Cluster{}, "LocationOfOrigin"),
 		cmpopts.IgnoreFields(clientcmdapi.Context{}, "LocationOfOrigin"),
+		cmpopts.IgnoreFields(clientcmdapi.Cluster{}, "LocationOfOrigin"),
 		cmpopts.IgnoreFields(clientcmdapi.AuthInfo{}, "LocationOfOrigin"),
 		cmpopts.IgnoreFields(clientcmdapi.AuthInfo{}, "Exec"),
-	)
-
-	t.Run("clearKubeconfig", func(t *testing.T) {
-		err := clearKubeconfig()
-		assert.NilError(t, err)
+	}
 
-		expected := clientcmdapi.Config{
-			Clusters:  map[string]*clientcmdapi.Cluster{},
-			Contexts:  map[string]*clientcmdapi.Context{},
-			AuthInfos: map[string]*clientcmdapi.AuthInfo{},
+	t.Run("OneNamespace", func(t *testing.T) {
+		expectedContexts := map[string]*clientcmdapi.Context{
+			"infra:connected": {
+				AuthInfo:  "user",
+				Cluster:   "infra:connected",
+				Namespace: "namespace",
+			},
 		}
 
-		actual, err := clientConfig().RawConfig()
-		assert.NilError(t, err)
+		actual := run(t, api.Grant{Resource: "connected.namespace"})
 
-		assert.DeepEqual(t, expected, actual, cmpopts.EquateEmpty())
+		assert.DeepEqual(t, actual.Contexts, expectedContexts, kubeConfigCmpOpts...)
+		assert.DeepEqual(t, actual.Clusters, expectedClusters, kubeConfigCmpOpts...)
+		assert.DeepEqual(t, actual.AuthInfos, expectedAuthInfos, kubeConfigCmpOpts...)
 	})
-}
 
-func TestWriteKubeconfig_UserNamespaceOverride(t *testing.T) {
-	home := t.TempDir()
-	t.Setenv("HOME", home)
-	t.Setenv("USERPROFILE", home)
+	t.Run("MultipleNamespaces", func(t *testing.T) {
+		expectedContexts := map[string]*clientcmdapi.Context{
+			"infra:connected": {
+				AuthInfo:  "user",
+				Cluster:   "infra:connected",
+				Namespace: "namespace",
```

Comment:
I thought the expectation would be for `Namespace: ""`, since this user has access to multiple namespaces. Is this test expecting the right thing?

Reply:
`Namespace: ""` is equivalent to `Namespace: "default"` which the user may not have access to. It makes sense to set `Namespace` to something as long as the user doesn't have some form of global access. We can probably introduce heuristics to make the selection smarter but for now, it'll pick the first namespace user has access to.

Reply:
I guess if one of the namespaces they have access to is default, then it would make sense for that to be the namespace. If they don't have access to default then any arbitrary one seems reasonable.